### PR TITLE
[qat] ConvReLU2d module

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack:
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #23022 [qat] LinearReLU module&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D16286800/)
&nbsp;&nbsp;&nbsp;&nbsp;:black_circle:&nbsp; **#23008 [qat] ConvReLU2d module**&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D16286670/)

Added ConvReLU2d module to convert from nn._intrinsic.ConvReLU2d to nn._intrinsic.qat.ConvReLU2d

Differential Revision: [D16286670](https://our.internmc.facebook.com/intern/diff/D16286670/)

### CODE REVIEW

File: torch/nn/_intrinsic/qat/modules/conv_relu.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                 activation_fake_quant=default_qat_qconfig.activation(),
+                 weight_fake_quant=default_qat_qconfig.weight()):
+        super(ConvReLU2d, self).__init__(in_channels, out_channels, kernel_size,
+                                         stride=stride, padding=padding, dilation=dilation,
+                                         groups=groups, bias=bias, padding_mode=padding_mode)
+        self.observer = activation_fake_quant
+        self.weight_fake_quant = weight_fake_quant
+
+    def forward(self, input):
+        return self.observer(F.relu(conv2d_forwad(input, self.padding_mode,
```
Comment (vadimkantorov) [inline torch/nn/_intrinsic/qat/modules/conv_relu.py]: nit: typo `conv2d_forwad` instead of `conv2d_forward`?

File: torch/nn/_intrinsic/qat/modules/conv_relu.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+from torch.nn.qat import Conv2d as QATConv2d
+from torch.nn.modules.conv import conv2d_forward
+from torch.quantization.QConfig import default_qat_qconfig
+import torch.nn.functional as F
+
+class ConvReLU2d(QATConv2d):
+    r"""
+    A Conv2d module attached with FakeQuantize modules for both output
```
Comment (jianyuh) [inline torch/nn/_intrinsic/qat/modules/conv_relu.py]: A ConvReLU2d module (Conv2d fused with ReLU)

**Comment by vadimkantorov** [comment]
A naming question: has Conv2dReLU been considered? It's better if more fused ops are introduced like Conv2dBatchNormReLUDropout - it's more predictable for the user for search in the docs: just append a few module names together.

**Comment by jerryzh168** [comment]
@vadimkantorov yes, reason is BatchNorm has 2d as well, I feel it's a bit redundant to have Conv2dBatchNorm2dReLU. so we have ConvBn2d and ConvBnReLU2d.

File: torch/nn/_intrinsic/qat/modules/conv_relu.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    We adopt the same interface as `torch.nn.Conv2d`, please see
+    https://pytorch.org/docs/stable/nn.html?highlight=conv2d#torch.nn.Conv2d
+    for documentation.
+
+    Similar to `torch.nn.Conv2d`, with FakeQuantize modules initialized to
+    default.
+
+    Attributes:
+        observer: fake quant module for output activation, it's called observer
+            to align with post training flow, TODO: rename?
```
Comment (dzhulgakov) [inline torch/nn/_intrinsic/qat/modules/conv_relu.py]: remove TODO?

File: torch/nn/_intrinsic/qat/modules/conv_relu.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+    def __init__(self, in_channels, out_channels, kernel_size, stride=1,
+                 padding=0, dilation=1, groups=1,
+                 bias=True, padding_mode='zeros',
+                 activation_fake_quant=default_qat_qconfig.activation,
+                 weight_fake_quant=default_qat_qconfig.weight):
+        super(ConvReLU2d, self).__init__(in_channels, out_channels, kernel_size,
+                                         stride=stride, padding=padding, dilation=dilation,
+                                         groups=groups, bias=bias, padding_mode=padding_mode)
+        self.observer = activation_fake_quant()
```
Comment (raghuramank100) [inline torch/nn/_intrinsic/qat/modules/conv_relu.py]: Lets settle on a name: Observer is not correct. How about calling it 'activation_post_process': This can be either observers or fake quant.

**Review by raghuramank100 — APPROVED** [review]
Approved, pending addressing of comment

File: torch/nn/_intrinsic/qat/modules/conv_relu.py (hunk 4 above)
Comment (jerryzh168) [reply torch/nn/_intrinsic/qat/modules/conv_relu.py]: OK, we can change it later after everything settles. since this requires changing all references to  ".observer"
